src/deadline_test_fixtures/deadline/resources.py

```diff
@@ -652,11 +831,390 @@ def __str__(self) -> str:  # pragma: no cover
         )
 
 
+@dataclass
+class DependencyCounts:
+    consumers_resolved: int
+    consumers_unresolved: int
+    dependencies_resolved: int
+    dependencies_unresolved: int
+
+    @staticmethod
+    def from_api_response(response: dict[str, Any]) -> DependencyCounts:
+        return DependencyCounts(
+            consumers_resolved=response["consumersResolved"],
+            consumers_unresolved=response["consumersUnresolved"],
+            dependencies_resolved=response["dependenciesResolved"],
+            dependencies_unresolved=response["dependenciesUnresolved"],
+        )
+
+
+@dataclass
+class Step:
+    farm: Farm
+    queue: Queue
+    job: Job
+    id: str
+
+    name: str
+    created_at: datetime.datetime
+    created_by: str
+    lifecycle_status: Literal[
+        "CREATE_COMPLETE", "UPDATE_IN_PROGRESS", "UDATE_FAILED", "UPDATE_SUCCEEDED"
+    ]
```

Comment:
Typo: `UDATE_FAILED` -> `UPDATE_FAILED`


Likewise, should we change this to use `StepLifecycleStatus`?

Reply:
Switched to `StepLifecycleStatus`